public/app/features/alerting/unified/api/prometheusApi.ts

```diff
@@ -22,9 +23,14 @@ interface PromRulesOptions {
   groupNextToken?: string;
 }
 
+type GrafanaPromRulesOptions = Omit<PromRulesOptions, 'ruleSource'> & {
+  dashboardUid?: string;
+  panelId?: number;
+};
+
 export const prometheusApi = alertingApi.injectEndpoints({
   endpoints: (build) => ({
-    groups: build.query<PromRulesResponse, PromRulesOptions>({
```

Comment:
Are we able to make this error if you try and use it with `ruleSource.uid === 'grafana'`? So we don't accidentally try and use it instead of `getGrafanaGroups`?

Reply:
Good question. Technically, nothing terrible happens if you use this endpoint with `grafana`. You'll just get a more narrow group and rule type (Prometheus types are subset of Grafana types here) and you might use wrong query parameters (because there is no 1:1 match) 

I wanted to have one endpoint, and do some Typescript magic to get correct types but I haven't found a way to do that, that's why there are two endpoints

Reply:
I found this approach, but I couldn't get it to work properly with the endpoints and I think it'd be overkill anyway :)

```tsx
type ReservedNames = "grafana"

function doStuff<T extends string>(
  v: T & (T extends ReservedNames ? "Use getGrafanaGroups instead" : {}),
) {
  return v;
}

doStuff("grafana");
doStuff("something else");
```

Reply:
I don't think we proposed solution above is very elegant considering there aren't any proper semantics for [negated types](https://github.com/microsoft/TypeScript/issues/4196).

We could throw an error on runtime to prevent the endpoint from being called with `uid === grafana`, and write a unit test + integration test for it?

Reply:
Yup I wasn't suggesting we necessarily use that approach - it is overkill and we could just cover it with an error + unit test

Reply:
FYI I tried forcing the method to throw an error, and the error boundary doesn't catch the problem properly - something to watch for. It catches some errors but the async ones via the RTK endpoints + flowing through all the generators code won't trigger it